### DISCUSSION TITLE
Disable proactive prompt to switch network

### DISCUSSION THREAD

**Discussion by brunowowk** [family/connectkit#106] [Ideas]
Usually the user's connected Network doesn't matter up to the point he decides to take some on-chain action. This is the case on mint pages, for instance, where the user should be able to check the artwork, read a bit of the details, and then just be prompted to switch Networks when the time to mint comes.

Having the switch network prompt pop up when the page is open is disruptive.

It would be great to have the option to prompt for a Network change only when the user tries to use their wallet to sign something, instead of whenever the user is connected to the dapp

**Comment by ryanberckmans** [comment]
Totally agree with this suggestion:

Right now, if the user's wallet is connected and has a network selected that's unsupported by the page's wagmi client, connectkit pops up a modal that forces the user to change their wallet's active network to a supported network. Today, this modal can't be dismissed or configured.

For various UX reasons and use cases, this modal should be dismissable and configurable to be disabled.

As a detail, connectkit should likely have no reliance on the rpc endpoint associated with the user's connected wallet's current active network because it should use only rpc endpoints provided to the wagmi client. So, afaict, there should be no technical blocker in simply ignoring whatever network the user's wallet current has active until later in the UX (eg. transaction signing).

**Comment by JamieLottering** [comment]
I believe this is possible by setting `initialChainId` to `0`

```typescript
<ConnectKitProvider options={{ initialChainId: 0 }}>
```

see: https://docs.family.co/connectkit/api-reference#ConnectKitProvider-initialChainId

**Reply by brunowowk** [reply]
With that setting, it only delays the switch network prompt until a wallet is connected.

If the user gets back to the app after connecting for the first time, he is prompted to switch network as soon as the app is opened, of if the network is changed.

The proposed UX here is: do not ask to switch the network until an action is requested from the user (signing message, sending tx, etc)

**Reply by JamieLottering** [reply]
That makes sense thanks for clarifying. Totally agree- that's what made me search out that option in the first place. Changing network when I don't need to is jarring

**Comment by lochie** [comment]
Have started a PR regarding this!
https://github.com/family/connectkit/pull/129

**Reply by brunowowk** [reply]
Thanks for acting so quick on the feedback. Such an awesome team! 😄 

Can't wait to see the wallet

**Reply by lochie** [reply]
PR has been merged and will be available in the next release ✨
Thanks @volkyeth!
